cli/api/commands/install.ts

```diff
@@ -6,37 +6,88 @@ import { promisify } from "util";
 import * as childProcess from "child_process";
 import { dataform } from "df/protos/ts";
 
-export async function install(projectPath: string, skipInstall?: boolean) {
-  if (skipInstall) {
-    return;
-  }
+export const MISSING_CORE_VERSION_ERROR =
+  "dataformCoreVersion must be specified either in workflow_settings.yaml or via a package.json";
+
+export async function install(projectPath: string) {
   const resolvedProjectPath = path.resolve(projectPath);
-  const workflowSettingsPath = path.join(resolvedProjectPath, "workflow_settings.yaml");
   const packageJsonPath = path.join(resolvedProjectPath, "package.json");
-  const packageLockJsonPath = path.join(resolvedProjectPath, "package-lock.json");
-
-  let installCommand = "npm i --ignore-scripts";
 
   // Core's readWorkflowSettings method cannot be used for this because Core assumes that
   // `require` can read YAML files directly.
-  const dataformCoreVersion = readDataformCoreVersionIfPresent(workflowSettingsPath);
-
+  const dataformCoreVersion = readDataformCoreVersionFromWorkflowSettings(resolvedProjectPath);
   if (dataformCoreVersion) {
-    // If there are other packages already in the package.json, specifying a specific package to
-    // install will trigger the other packages to be installed too.
-    installCommand += ` @dataform/core@${dataformCoreVersion}`;
+    throw Error(
+      "The install method cannot be used when dataformCoreVersion is managed by a " +
+        "workflow_settings.yaml file; packages will be installed at compile time instead"
+    );
   }
 
-  if (!dataformCoreVersion && !fs.existsSync(packageJsonPath)) {
-    throw new Error(
-      "dataformCoreVersion must be specified either in workflow_settings.yaml or via a package.json"
+  if (!fs.existsSync(packageJsonPath)) {
+    throw new Error(MISSING_CORE_VERSION_ERROR);
+  }
+
+  await promisify(childProcess.exec)("npm i --ignore-scripts", { cwd: resolvedProjectPath });
+}
+
+export async function runInstallIfWorkflowSettingsDataformCoreVersion(
+  projectPath: string
+): Promise<string> {
+  const resolvedProjectPath = path.resolve(projectPath);
+  const packageJsonPath = path.join(resolvedProjectPath, "package.json");
+  const packageLockJsonPath = path.join(resolvedProjectPath, "package-lock.json");
+
+  const workflowSettingsDataformCoreVersion = readDataformCoreVersionFromWorkflowSettings(
+    resolvedProjectPath
+  );
+  if (!fs.existsSync(packageJsonPath)) {
+    if (!workflowSettingsDataformCoreVersion) {
+      throw new Error(MISSING_CORE_VERSION_ERROR);
+    }
+
+    fs.writeFileSync(
+      packageJsonPath,
+      `{
+  "dependencies":{
+  "@dataform/core": "${workflowSettingsDataformCoreVersion}"
+  }
+}`
     );
+    fs.writeFileSync(
+      packageLockJsonPath,
+      `{
+  "requires": true,
+  "dependencies": {
+    "@dataform/core": {
+      "version": "${workflowSettingsDataformCoreVersion}"
+    }
   }
+}`
+    );
```

Comment:
why bother writing this file?

- the files will be deleted anyway
- `workflowSettingsDataformCoreVersion` should be a specific version anyway, so `package.json` is enough
- if `workflowSettingsDataformCoreVersion` isn't a specific version (i.e. semver), then this would be invalid

Reply:
To mimic internally, the `npm i` below should be `npm ci`, which requires a `package-lock.json` too - have updated

Reply:
From chat - we can remove `npm ci` from the server for this case too instead, which reduces overall complexity too